Spells/Druid: Implement Astral Communion talent with Celestial Alignment

aquadeus opened the pull request:
<!--- (**********************************)
      (** Fill in the following fields **)
      (**********************************) --->

**Changes proposed:**

-  Implement Astral Communion talent energize with Celestial Alignment

**Tests performed:**

Tested in-game


<!--- Notes
- Enable the setting "[√] Allow edits from maintainers." when creating your pull request.
- If this PR only contains SQL files, open a new issue instead and post or link the SQL in the issue.
- When adding new SQL files, name them 9999_99_99_99_db_name.sql to reduce the chance of possible merge conflicts.
--->

aquadeus committed:
Spells/Druid: Implement Astral Communion talent

Shauren commented:
This one is incorrect, entering eclipse with Celestial Alignment/Incarnation should only energize once, not twice

Shauren reviewed (changes requested):
This one is incorrect, entering eclipse with Celestial Alignment/Incarnation should only energize once, not twice

aquadeus committed:
Merge branch 'master' of https://github.com/TrinityCore/TrinityCore into druid_astral_communion

aquadeus committed:
Revert code deleted by mistake on merge

aquadeus committed:
Merge branch 'master' of https://github.com/TrinityCore/TrinityCore into druid_astral_communion

aquadeus committed:
Fix astral communion energize with Celestial Alignment

aquadeus commented:
ok i updated the pr and fixed the double energize (energize happens when wrath or starfire procs the eclipse) with celestial alignment, now missing the energize part with incarnation of elune, but that will be in another pr

aquadeus committed:
Missing validate

Shauren committed:
Merge branch 'master' into aquadeus_druid_astral_communion

Shauren committed:
prepare merge
